### DISCUSSION TITLE
How to judge the specific function call?

### DISCUSSION THREAD

**Discussion by j10ccc** [typst/typst#2071] [Q&A] [answered]
I have a function to render some content, and want to judge **the first function call**.

I try to implement it by getting `integer` value from `counter` instance, but there is no related api?

```typst
#let render_counter = counter("render")

#let render(body, title: "") = [
  #render_counter.step()

  // FIXME: how to get integer or another way to judge the condition
  #if render_counter.display() == 1 [

  ] else [

  ]
]
```

**Comment by Dherse** [comment, marked as answer]
You need to use the `at` [method](https://typst.app/docs/reference/meta/counter/#methods-at) on `counter`. Inside of a `locate` [block](https://typst.app/docs/reference/meta/locate/). This is all covered in the documentation in the [time travel section](https://typst.app/docs/reference/meta/counter/#time-travel).

**Reply by Dherse** [reply]
To give an example:

```ts
#let render_counter = counter("render")

#let render(body, title: "") = locate(loc => {
  render_counter.step()

  // FIXME: how to get integer or another way to judge the condition
  if render_counter.at(loc) == 1 [

  ] else [

  ]
})
```

**Reply by laurmaedje** [reply]
Note that this will not respect the step because `loc` is before the step. The step should probably be generated outside of locate.

**Reply by Dherse** [reply]
Yes, I meant to mention it, but I was not sure what was @j10ccc's intension

**Comment by j10ccc** [comment]
OK, I've referred @Dherse example. I just want to specially render the first component.

I modify your example a little of the `if` statement and achieve my goals.

```js
render_counter.step()
if render_counter.at(loc).at(0) == 0 [
  this is the first paragraph.
] else [
  this is not the first paragraph.
]
```

Before starting the discussion, I have read the [time travel section](https://typst.app/docs/reference/meta/counter/#time-travel). To be honest, it's a little obscure.
